chain/chunks/src/lib.rs

```diff
@@ -1339,20 +1440,78 @@ impl ShardsManager {
         );
         store_update.commit()?;
 
+        // Merge parts and receipts included in the partial encoded chunk into chunk cache
         self.encoded_chunks.merge_in_partial_encoded_chunk(partial_encoded_chunk);
 
-        // Forward my parts to others tracking this chunk's shard
+        // 3. Process the forwarded parts in chunk_forwards_cache
+        match self.chunk_forwards_cache.cache_remove(&chunk_hash) {
+            None => {}
+            Some(forwarded_parts) => {
+                // We have the header now, and there were some parts we were forwarded earlier.
+                // Let's process those parts now.
+                let forwarded_chunk = PartialEncodedChunkV2 {
+                    header: header.clone(),
+                    parts: forwarded_parts.into_iter().map(|(_, part)| part).collect(),
+                    receipts: Vec::new(),
+                };
+                // Make sure we mark this chunk as being requested so that it is handled
+                // properly in the next call. Note no requests are actually sent at this point.
+                self.request_chunk_single_mark_only(header);
+                return self.process_partial_encoded_chunk(
```

Comment:
I'd mention that here we call ourselves recursively, basically "simulating" as that forwarded part was just received from the network